src/main/kotlin/traffic_simulation/Vehicle.kt

```diff
@@ -18,6 +18,7 @@ class Vehicle(val id: Int, val wannaDriveInHours: MutableList<Int>) {
             delay += 1
             this.delayedInHours.add(timestep)
             this.delayedInHours.sort()
+            this.wannaDriveInHours.add(timestep + 1)
```

Comment:
Mit der Zeile können Verspätungen doppelt gezählt werden:

* `delay` wird hochgesetzt und muss nachher wieder abgebaut werden
* außerdem wird jetzt u.U. an einer Stunden gefahren, an der es vorher nicht so war.

Hier sind Tests (können in `VehicleTest`), die das unterstreichen:

```kotlin
@Test
fun delayAndDrive_mixedDelaysAndDriveRequirements_driveRequirementsCorrectlyReported() {
    val tesla = Vehicle(1, mutableListOf(1, 5))
    assertTrue(tesla.vehicleWantsToDriveAt(1))

    tesla.getDelayedAtHour(1)
    // couldn't drive at 1, so wants to drive at 2
    assertTrue(tesla.vehicleWantsToDriveAt(2))

    tesla.getDelayedAtHour(2)
    // again, couldn't drive,  so wants to drive at 3
    assertTrue(tesla.vehicleWantsToDriveAt(3))

    tesla.driveAtHour(3)
    // finally time to drive; this resolves delay from hour 1, so no more driving
    assertFalse(tesla.vehicleWantsToDriveAt(4))

    // wants to drive at hour 5 as per original requirements
    assertTrue(tesla.vehicleWantsToDriveAt(5))
}

@Test
fun delayAndDrive_mixedDelaysAndDriveRequirements_driveRequirementsCarriedAcross() {
    val tesla = Vehicle(1, mutableListOf(1, 2, 4))
    assertTrue(tesla.vehicleWantsToDriveAt(1))

    tesla.getDelayedAtHour(1)
    // couldn't drive at 1 _and_ wanted to drive at 2 anyways
    assertTrue(tesla.vehicleWantsToDriveAt(2))

    tesla.getDelayedAtHour(2)
    // couldn't drive at 1 & 2, so definitely wants to drive at 3
    assertTrue(tesla.vehicleWantsToDriveAt(3))

    tesla.driveAtHour(3)
    // still one hour delayed _and_ wants to drive at 4 anyways
    assertTrue(tesla.vehicleWantsToDriveAt(4))

    tesla.driveAtHour(4)
    // driving at 4 accomplished but still one hour delayed, so wants to drive again
    assertTrue(tesla.vehicleWantsToDriveAt(5))

    tesla.driveAtHour(5)
    // finally all done
    assertFalse(tesla.vehicleWantsToDriveAt(6))
}
```

Die scheitern mit der Zeile, gehen ohne aber durch. Ich habe mir die anderen Tests nicht genau angeguckt, gehe aber davon aus, dass sie fehlerhaft sind.